framework/test/gtest/ServiceHooksTest.cpp

```diff
@@ -305,6 +292,54 @@ namespace
 
 } // end unnamed namespace
 
+class MockServiceEventListenerHook : public ServiceEventListenerHook
+{
+  public:
+    MockServiceEventListenerHook(int id, BundleContext const& context)
+        : fake_(TestServiceEventListenerHook(id, context))
```

Comment:
I don't think this is correct. The `id` and `fake_` members are not needed for the mock and for testing the ordering of callbacks.

Reply:
See below